v1/ratelimit/redis_delayed_sync_pipelined_test.go

```diff
@@ -460,3 +461,52 @@ func Test_GetResetAt_DefaultIsZero_Pipelined(t *testing.T) {
 		t.Fatalf("expected resetAt to be 0 for unused key, got %d", got)
 	}
 }
+
+// Note: this test will most of the time pass due to the data race it tests against being very hard to replicate
+// without introducing artificial sleeps. A fix has been written along with this test, so this is more of a reference
+// for future devs rather than a useful and rigorous test.
+func TestNewlyJoinedClientSyncsSameTimeAsFirstRequest(t *testing.T) {
+	/**
+	1. HTTP gateways are rate-limiting user as per normal
+	2. Suddenly, user authenticates in WS gateway
+	3. User makes WS request
```

Comment:
I think ws/http gateways are not terms valid for the library. The term used here is Client.